src/org/skypro/skyshop/searchengine/SearchEngine.java

```diff
@@ -14,11 +15,11 @@ public SearchEngine() {
         this.searchableArray = new ArrayList<>();
     }
 
-    public List<Searchable> search(String substring) throws BestResultNotFoundException {
-        List<Searchable> searchArray = new ArrayList<>();
+    public TreeMap<String, Searchable> search(String substring) throws BestResultNotFoundException {
```

Comment:
не забываем про полиморфизм
```suggestion
    public Map<String, Searchable> search(String substring) throws BestResultNotFoundException {
```